### PR TITLE
Added wallet functions

### LINKED ISSUE #8: Function for creating accounts in wallets
We discussed to create a new function
```go
prnm.CreateAccount(walletPath, passphrase string) (account string, err error)
```
that opens or creates a wallet path and then generates a new account in it and returns its address.

_Originally posted by @sebastianst in https://github.com/perun-network/perun-eth-mobile/pull/1/review_comment/create_

### PR DESCRIPTION
**Pull request by ggwpez**
Added
+ `CreateAccount(keystorePath, password string) (*Address, error)`
+ `ImportAccount(keystorePath, password, secretKey string) (*Address, error)`

### CODE REVIEW

**Review by sebastianst — CHANGES_REQUESTED** [review]
We now work at three different places with a `walletPath, Passphrase` tuple that stands for a wallet, so I think it's time for a wallet wrapper:
- Create `type Wallet { w ethwallet.Wallet }`
- with constructor `NewWallet(path, passphrase string) (*Wallet, error)` and
- with method `CreateAccount() *Address` and
- with method `ImportAccount(hexSecretKey string) error` and
- change `NewClient` to also include a `*Wallet` parameter and
- remove string fields `walletPath` and `passphrase` from type `Config`.

**Comment by ggwpez** [comment]
Using the standard `scrypt` parameters on Android is slower than on PC and since it has to unlock the wallet twice at startup right now, the App crashes after 9 seconds of startup time. Steffen said that Android probably stops Apps when they dont react withing a given time after startup.  
So i need to fix the Java app before being able to test this with standard parameter.

**Comment by sebastianst** [comment]
If you implement the wallet wrapper type as suggested in my review comment, there's only one unlock happening. Also, you changed the parameters from`2, 1` to the `keystore.StandardScrypt(N,P)` standard ones, which causes the long startup time now.

I think we should just use 2, 1 here for the time being. If this code is ever to be used in production, we make the parameters configurable.

File: wallet.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	acc, err := w.Unlock(&cfg.Address.addr)
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "unlocking account")
+	}
+	return w, acc.(*ethwallet.Account), nil
+}
+
+func createWallet(keystorePath, password string) (*ethwallet.Wallet, error) {
+	ks := keystore.NewKeyStore(keystorePath, keystore.StandardScryptN, keystore.StandardScryptP)
```
Comment (sebastianst) [inline wallet.go]: Use 2, 1 here for now...

File: client.go [hunk 2]
```diff
@@ -31,7 +28,7 @@ type Client struct {
 
 	client *client.Client
 
-	w       *ethwallet.Wallet
+	w       *Wallet
```
Comment (sebastianst) [inline client.go]: After `NewWallet`, you only need the `ethwallet.Wallet`, so I think you can keep `w` as `ethwallet.Wallet` as it was before.

File: wallet.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+)
+
+// Wallet represents an ethereum wallet. It uses the go-ethereum keystore to
+// store keys. Accessing the wallet is threadsafe, however you should not
+// create two wallets from the same key directory.
+// ref https://pkg.go.dev/perun.network/go-perun/backend/ethereum/wallet?tab=doc#Wallet
+type Wallet struct {
+	w              *ethwallet.Wallet
+	path, password string
```
Comment (sebastianst) [inline wallet.go]: You don't need the `path` in `Wallet`.

File: wallet.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		return nil, errors.WithMessage(err, "unlocking account")
+	}
+	return &Address{*(acc.Address()).(*ethwallet.Address)}, err
+}
+
+// CreateAccount returns the Address of a new randomly created Account.
+// ref https://pkg.go.dev/perun.network/go-perun/backend/ethereum/wallet?tab=doc#Wallet.NewAccount
+func (w *Wallet) CreateAccount(keystorePath, password string) (*Address, error) {
+	return &Address{ethwallet.Address(w.w.NewAccount().Account.Address)}, nil
```
Comment (sebastianst) [inline wallet.go]:
```suggestion
func (w *Wallet) CreateAccount() *Address {
	return &Address{ethwallet.Address(w.w.NewAccount().Account.Address)}
```

**Review by sebastianst — CHANGES_REQUESTED** [review]

File: wallet.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	acc, err := w.w.Unlock(wAcc.Address())
+	if err != nil {
+		return nil, errors.WithMessage(err, "unlocking account")
+	}
+	return &Address{*(acc.Address()).(*ethwallet.Address)}, err
+}
+
+// CreateAccount returns the Address of a new randomly created Account.
+// ref https://pkg.go.dev/perun.network/go-perun/backend/ethereum/wallet?tab=doc#Wallet.NewAccount
+func (w *Wallet) CreateAccount(keystorePath, password string) *Address {
```
Comment (sebastianst) [inline wallet.go]:
```suggestion
func (w *Wallet) CreateAccount() *Address {
```
Was already in last review...

**Review by sebastianst — CHANGES_REQUESTED** [review]

File: wallet.go (hunk 5 above)
Comment (ggwpez) [reply wallet.go]: Golang did not even show me a warning about that unused parameter…

Comment (sebastianst) [reply wallet.go]: I think `go lint` would. We should setup the vet/lint/spell/test pipeline soon...

File: wallet.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	acc, err := w.w.Unlock(wAcc.Address())
+	if err != nil {
+		return nil, errors.WithMessage(err, "unlocking account")
+	}
+	return &Address{*(acc.Address()).(*ethwallet.Address)}, err
+}
+
+// CreateAccount returns the Address of a new randomly created Account.
+// ref https://pkg.go.dev/perun.network/go-perun/backend/ethereum/wallet?tab=doc#Wallet.NewAccount
+func (w *Wallet) CreateAccount(password string) *Address {
```
Comment (sebastianst) [inline wallet.go]:
```suggestion
func (w *Wallet) CreateAccount() *Address {
```
Was that a test?

**Review by sebastianst — CHANGES_REQUESTED** [review]

**Review by sebastianst — APPROVED** [review]

**Pull request merged**